crates/cli/local_server.rs

```diff
@@ -0,0 +1,171 @@
+                    } else {
+                        (true, None)
+                    }
+                } else {
+                    (false, None)
+                }
+            };
+
+            let show_validity = {
+                let show_validity_maybe: Result<bool, ParseBoolError> = bool::from_str(show_validity.as_str());
+                if let Ok(provided_validity) = show_validity_maybe {
+                    Some(provided_validity)
+                } else {
+                    None
+                }
+            };
+
+            let port = {
+                let port_maybe: Result<i32, ParseIntError> = i32::from_str(gateway_port.as_str());
+                if let Ok(provided_port) = port_maybe {
+                    Some(provided_port)
+                } else {
+                    None
+                }
+            };
+
+            let cache = {
+                let maybe_cache: Result<bool, ParseBoolError> = bool::from_str(cache.as_str());
+                if let Ok(provided_cache) = maybe_cache {
+                    Some(provided_cache)
+                } else {
+                    None
+                }
+            };
+
+            let show_errors = {
+                let maybe_show_errors: Result<bool, ParseBoolError> = bool::from_str(show_errors.as_str());
+                if let Ok(provided_show_errors) = maybe_show_errors {
+                    Some(provided_show_errors)
+                } else {
+                    None
+                }
+            };
+
+            let mut response_result: Option<Response<Body>> = None;
+
+            if contract_id.is_none() {
+                response_result = Some(build_error("contractId was not provided in query parameters. A contract id must be provided."));
+            } else if height_invalid {
+                response_result = Some(build_error("The height provided is incorrect. Please provide a valid numeric value."));
+            } else if port.is_none() {
+                response_result = Some(build_error("The gateway port provided is invalid. Please provided a valid numeric value."));
+            } else if show_validity.is_none() {
+                response_result = Some(build_error("The 'showValidity' parameter contains wrong data. Please provided a valid boolean value."));
+            } else if cache.is_none() {
+                response_result = Some(build_error("The 'cache' parameter contains wrong data. Please provided a valid boolean value."));
+            } else if show_errors.is_none() {
+                response_result = Some(build_error("The 'showErrors' parameter contains wrong data. Please provided a valid boolean value."));
+            } else {
+                 let arweave = Arweave::new(port.unwrap(), gateway_host.to_owned(), gateway_protocol.to_owned(), ArweaveCache::new());
+                 let execute_result = execute_contract(&arweave, contract_id.unwrap().to_owned(), None, None, height, cache.unwrap(), show_errors.unwrap()).await;
```

Comment:
This line right here, when i comment it, it works. When I don't comment it, this is what happens:

[stacktrace.txt](https://github.com/three-em/3em/files/7878896/stacktrace.txt)


In summary: ```error[E0277]: `Rc<RefCell<three_em_js::HeapLimitState>>` cannot be sent between threads safely
   --> crates/cli/local_server.rs:166:36
    |
166 |   let server = Server::bind(&addr).serve(service);
    |                                    ^^^^^ `Rc<RefCell<three_em_js::HeapLimitState>>` cannot be sent between threads safely
    |
    = help: within `hyper::proto::h2::server::H2Stream<impl std::future::Future, Body>`, the trait `std::marker::Send` is not implemented for `Rc<RefCell<three_em_js::HeapLimitState>>`
    = note: required because it appears within the type `three_em_js::Runtime`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6, 't7, 't8, 't9, 't10, 't11, 't12> {ResumeTy, std::string::String, LoadedContract, Vec<GQLEdgeInterface>, IndexMap<std::string::String, bool>, std::option::Option<serde_json::Value>, bool, [closure@execute_contract::{closure#0}::{closure#0}], &'r Arweave, TransactionData, (i32, std::string::String, std::string::String), ContractType, serde_json::Value, Vec<u8>, Result<std::string::String, FromUtf8Error>, &'s str, &'t0 std::string::String, &'t1 (i32, std::string::String, std::string::String), impl for<'t2> std::future::Future, (), three_em_js::Runtime, std::vec::IntoIter<GQLEdgeInterface>, GQLEdgeInterface, GQLNodeInterface, &'t3 mut three_em_js::Runtime, impl for<'t4> std::future::Future, Result<std::option::Option<three_em_js::CallResult>, anyhow::Error>, std::option::Option<std::string::String>, impl for<'t5> std::future::Future, &'t6 [u8], &'t7 Vec<u8>, Cow<'t8, str>, &'t9 Cow<'t10, str>, three_em_evm::storage::Storage, impl for<'t11, 't12> std::future::Future}`
    = note: required because it appears within the type `[static generator@raw_execute_contract<[closure@execute_contract::{closure#0}::{closure#0}]>::{closure#0}]`
    = note: required because it appears within the type `from_generator::GenFuture<[static generator@raw_execute_contract<[closure@execute_contract::{closure#0}::{closure#0}]>::{closure#0}]>`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6> {ResumeTy, &'r Arweave, std::string::String, std::option::Option<std::string::String>, std::option::Option<usize>, bool, (tokio::future::maybe_done::MaybeDone<impl for<'s> std::future::Future>, tokio::future::maybe_done::MaybeDone<impl for<'t0> std::future::Future>), [closure@execute_contract::{closure#0}::{closure#3}], tokio::future::poll_fn::PollFn<[closure@execute_contract::{closure#0}::{closure#3}]>, (), Result<LoadedContract, anyhow::Error>, Result<(Vec<GQLEdgeInterface>, usize, bool), anyhow::Error>, LoadedContract, Vec<GQLEdgeInterface>, usize, IndexMap<std::string::String, bool>, std::option::Option<serde_json::Value>, &'t5 std::string::String, [closure@execute_contract::{closure#0}::{closure#0}], impl for<'t6> std::future::Future}`
    = note: required because it appears within the type `[static generator@execute_contract::{closure#0}]`
    = note: required because it appears within the type `from_generator::GenFuture<[static generator@execute_contract::{closure#0}]>`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6> {ResumeTy, Request<Body>, &'r Request<Body>, &'s Method, &'t0 Uri, &'t1 str, (&'t2 Method, &'t3 str), HashMap<std::string::String, std::string::String>, std::option::Option<std::string::String>, std::string::String, bool, std::option::Option<usize>, std::option::Option<bool>, std::option::Option<i32>, std::option::Option<Response<Body>>, Arweave, &'t4 Arweave, &'t5 std::string::String, impl for<'t6> std::future::Future, ()}`
    = note: required because it appears within the type `[static generator@crates/cli/local_server.rs:33:75: 159:2]`
    = note: required because it appears within the type `from_generator::GenFuture<[static generator@crates/cli/local_server.rs:33:75: 159:2]>`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `hyper::proto::h2::server::H2StreamState<impl std::future::Future, Body>`
    = note: required because it appears within the type `hyper::proto::h2::server::H2Stream<impl std::future::Future, Body>`
    = note: required because of the requirements on the impl of `hyper::common::exec::ConnStreamExec<impl std::future::Future, Body>` for `hyper::common::exec::Exec`

error[E0277]: `NonNull<deno_core::v8::Value>` cannot be sent between threads safely
   --> crates/cli/local_server.rs:166:36
    |
166 |   let server = Server::bind(&addr).serve(service);
    |                                    ^^^^^ `NonNull<deno_core::v8::Value>` cannot be sent between threads safely
    |
    = help: within `hyper::proto::h2::server::H2Stream<impl std::future::Future, Body>`, the trait `std::marker::Send` is not implemented for `NonNull<deno_core::v8::Value>`
    = note: required because it appears within the type `deno_core::v8::Global<deno_core::v8::Value>`
    = note: required because it appears within the type `three_em_js::Runtime`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6, 't7, 't8, 't9, 't10, 't11, 't12> {ResumeTy, std::string::String, LoadedContract, Vec<GQLEdgeInterface>, IndexMap<std::string::String, bool>, std::option::Option<serde_json::Value>, bool, [closure@execute_contract::{closure#0}::{closure#0}], &'r Arweave, TransactionData, (i32, std::string::String, std::string::String), ContractType, serde_json::Value, Vec<u8>, Result<std::string::String, FromUtf8Error>, &'s str, &'t0 std::string::String, &'t1 (i32, std::string::String, std::string::String), impl for<'t2> std::future::Future, (), three_em_js::Runtime, std::vec::IntoIter<GQLEdgeInterface>, GQLEdgeInterface, GQLNodeInterface, &'t3 mut three_em_js::Runtime, impl for<'t4> std::future::Future, Result<std::option::Option<three_em_js::CallResult>, anyhow::Error>, std::option::Option<std::string::String>, impl for<'t5> std::future::Future, &'t6 [u8], &'t7 Vec<u8>, Cow<'t8, str>, &'t9 Cow<'t10, str>, three_em_evm::storage::Storage, impl for<'t11, 't12> std::future::Future}`
    = note: required because it appears within the type `[static generator@raw_execute_contract<[closure@execute_contract::{closure#0}::{closure#0}]>::{closure#0}]`
    = note: required because it appears within the type `from_generator::GenFuture<[static generator@raw_execute_contract<[closure@execute_contract::{closure#0}::{closure#0}]>::{closure#0}]>`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6> {ResumeTy, &'r Arweave, std::string::String, std::option::Option<std::string::String>, std::option::Option<usize>, bool, (tokio::future::maybe_done::MaybeDone<impl for<'s> std::future::Future>, tokio::future::maybe_done::MaybeDone<impl for<'t0> std::future::Future>), [closure@execute_contract::{closure#0}::{closure#3}], tokio::future::poll_fn::PollFn<[closure@execute_contract::{closure#0}::{closure#3}]>, (), Result<LoadedContract, anyhow::Error>, Result<(Vec<GQLEdgeInterface>, usize, bool), anyhow::Error>, LoadedContract, Vec<GQLEdgeInterface>, usize, IndexMap<std::string::String, bool>, std::option::Option<serde_json::Value>, &'t5 std::string::String, [closure@execute_contract::{closure#0}::{closure#0}], impl for<'t6> std::future::Future}`
    = note: required because it appears within the type `[static generator@execute_contract::{closure#0}]`
    = note: required because it appears within the type `from_generator::GenFuture<[static generator@execute_contract::{closure#0}]>`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6> {ResumeTy, Request<Body>, &'r Request<Body>, &'s Method, &'t0 Uri, &'t1 str, (&'t2 Method, &'t3 str), HashMap<std::string::String, std::string::String>, std::option::Option<std::string::String>, std::string::String, bool, std::option::Option<usize>, std::option::Option<bool>, std::option::Option<i32>, std::option::Option<Response<Body>>, Arweave, &'t4 Arweave, &'t5 std::string::String, impl for<'t6> std::future::Future, ()}`
    = note: required because it appears within the type `[static generator@crates/cli/local_server.rs:33:75: 159:2]`
    = note: required because it appears within the type `from_generator::GenFuture<[static generator@crates/cli/local_server.rs:33:75: 159:2]>`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `hyper::proto::h2::server::H2StreamState<impl std::future::Future, Body>`
    = note: required because it appears within the type `hyper::proto::h2::server::H2Stream<impl std::future::Future, Body>`
    = note: required because of the requirements on the impl of `hyper::common::exec::ConnStreamExec<impl std::future::Future, Body>` for `hyper::common::exec::Exec`

error[E0277]: `NonNull<Isolate>` cannot be sent between threads safely
   --> crates/cli/local_server.rs:166:36
    |
166 |   let server = Server::bind(&addr).serve(service);
    |                                    ^^^^^ `NonNull<Isolate>` cannot be sent between threads safely
    |
    = help: within `hyper::proto::h2::server::H2Stream<impl std::future::Future, Body>`, the trait `std::marker::Send` is not implemented for `NonNull<Isolate>`
    = note: required because it appears within the type `OwnedIsolate`
    = note: required because it appears within the type `std::option::Option<OwnedIsolate>`
    = note: required because it appears within the type `JsRuntime`
    = note: required because it appears within the type `three_em_js::Runtime`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6, 't7, 't8, 't9, 't10, 't11, 't12> {ResumeTy, std::string::String, LoadedContract, Vec<GQLEdgeInterface>, IndexMap<std::string::String, bool>, std::option::Option<serde_json::Value>, bool, [closure@execute_contract::{closure#0}::{closure#0}], &'r Arweave, TransactionData, (i32, std::string::String, std::string::String), ContractType, serde_json::Value, Vec<u8>, Result<std::string::String, FromUtf8Error>, &'s str, &'t0 std::string::String, &'t1 (i32, std::string::String, std::string::String), impl for<'t2> std::future::Future, (), three_em_js::Runtime, std::vec::IntoIter<GQLEdgeInterface>, GQLEdgeInterface, GQLNodeInterface, &'t3 mut three_em_js::Runtime, impl for<'t4> std::future::Future, Result<std::option::Option<three_em_js::CallResult>, anyhow::Error>, std::option::Option<std::string::String>, impl for<'t5> std::future::Future, &'t6 [u8], &'t7 Vec<u8>, Cow<'t8, str>, &'t9 Cow<'t10, str>, three_em_evm::storage::Storage, impl for<'t11, 't12> std::future::Future}`
    = note: required because it appears within the type `[static generator@raw_execute_contract<[closure@execute_contract::{closure#0}::{closure#0}]>::{closure#0}]`
    = note: required because it appears within the type `from_generator::GenFuture<[static generator@raw_execute_contract<[closure@execute_contract::{closure#0}::{closure#0}]>::{closure#0}]>`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6> {ResumeTy, &'r Arweave, std::string::String, std::option::Option<std::string::String>, std::option::Option<usize>, bool, (tokio::future::maybe_done::MaybeDone<impl for<'s> std::future::Future>, tokio::future::maybe_done::MaybeDone<impl for<'t0> std::future::Future>), [closure@execute_contract::{closure#0}::{closure#3}], tokio::future::poll_fn::PollFn<[closure@execute_contract::{closure#0}::{closure#3}]>, (), Result<LoadedContract, anyhow::Error>, Result<(Vec<GQLEdgeInterface>, usize, bool), anyhow::Error>, LoadedContract, Vec<GQLEdgeInterface>, usize, IndexMap<std::string::String, bool>, std::option::Option<serde_json::Value>, &'t5 std::string::String, [closure@execute_contract::{closure#0}::{closure#0}], impl for<'t6> std::future::Future}`
    = note: required because it appears within the type `[static generator@execute_contract::{closure#0}]`
    = note: required because it appears within the type `from_generator::GenFuture<[static generator@execute_contract::{closure#0}]>`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `for<'r, 's, 't0, 't1, 't2, 't3, 't4, 't5, 't6> {ResumeTy, Request<Body>, &'r Request<Body>, &'s Method, &'t0 Uri, &'t1 str, (&'t2 Method, &'t3 str), HashMap<std::string::String, std::string::String>, std::option::Option<std::string::String>, std::string::String, bool, std::option::Option<usize>, std::option::Option<bool>, std::option::Option<i32>, std::option::Option<Response<Body>>, Arweave, &'t4 Arweave, &'t5 std::string::String, impl for<'t6> std::future::Future, ()}`
    = note: required because it appears within the type `[static generator@crates/cli/local_server.rs:33:75: 159:2]`
    = note: required because it appears within the type `from_generator::GenFuture<[static generator@crates/cli/local_server.rs:33:75: 159:2]>`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `impl std::future::Future`
    = note: required because it appears within the type `hyper::proto::h2::server::H2StreamState<impl std::future::Future, Body>`
    = note: required because it appears within the type `hyper::proto::h2::server::H2Stream<impl std::future::Future, Body>`
    = note: required because of the requirements on the impl of `hyper::common::exec::ConnStreamExec<impl std::future::Future, Body>` for `hyper::common::exec::Exec`

error[E0277]: `Rc<three_em_js::loader::EmbeddedModuleLoader>` cannot be sent between threads safely
   --> crates/cli/local_server.rs:166:36
    |
166 |   let server = Server::bind(&addr).serve(service);
    |                                    ^^^^^ `Rc<three_em_js::loader::EmbeddedModuleLoader>` cannot be sent between threads safely
```

but it goes way beyond that lol
@littledivy